### PR TITLE
Phoenix v1/tif

### PR DESCRIPTION
**Pull request by jarry-xiao**
Changed needed to support TIF feature. This only addresses the changes needed for the rust SDK

### CODE REVIEW

File: rust/phoenix-sdk-core/src/sdk_client_core.rs [hunk 1]
```diff
@@ -421,8 +424,50 @@ impl SDKClientCore {
...
+                        base_lots_removed,
+                    }) => market_events.push(PhoenixEvent {
+                        market: header.market,
+                        sequence_number: header.sequence_number,
+                        slot: header.slot,
+                        timestamp: header.timestamp,
+                        signature: *sig,
+                        signer: header.signer,
+                        event_index: index as u64,
+                        details: MarketEventDetails::Reduce(Reduce {
```
Comment (jarry-xiao) [inline rust/phoenix-sdk-core/src/sdk_client_core.rs:460]: @eugene-chen I'm not super happy about this as it merges the TIF cancel into a Reduce. It was the easiest way to get everything working without a major refactor. Happy to change it if necessary

File: rust/phoenix-sdk-core/src/packet_decoder.rs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+use phoenix::state::{SelfTradeBehavior, Side};
+
+#[derive(BorshDeserialize, BorshSerialize)]
+pub enum OrderPacketEnum {
+    PostOnly,
+    Limit,
+}
+
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct PostOnlyPacketDeprecated {
```
Comment (eugene-chen) [inline rust/phoenix-sdk-core/src/packet_decoder.rs:11]: prefer `DeprecatedPostOnlyPacket`, and also for current version to live on top of deprecated version, and also an aggressive doc saying that this shouldn't be used

File: rust/phoenix-sdk-core/src/sdk_client_core.rs (hunk 1 above)
Comment (eugene-chen) [reply rust/phoenix-sdk-core/src/sdk_client_core.rs:460]: tldr on what needs to be refactored? I definitely prefer to maintain the 1:1 mapping between `PhoenixMarketEvent` and `MarketEventDetails`

**Comment by eugene-chen** [comment]
also note in PR description that this is rust only for now. TS should go in separate PR

File: rust/phoenix-sdk-core/src/sdk_client_core.rs (hunk 1 above)
Comment (jarry-xiao) [reply rust/phoenix-sdk-core/src/sdk_client_core.rs:460]: You would need a new enum in MarketEventDetails. Reduce happens to encapsulate the information needed, but it's not super clean. This was the minimum work needed to get testing to work

File: rust/phoenix-sdk-core/src/packet_decoder.rs (hunk 2 above)
Comment (jarry-xiao) [reply rust/phoenix-sdk-core/src/packet_decoder.rs:11]: I can probably make this private. It's only used as helper for deserialization

**Review by eugene-chen — APPROVED** [review]

**Pull request merged**